cpp/benchmarks/ast/transform.cpp

```diff
@@ -86,38 +87,25 @@ static void BM_ast_transform(benchmark::State& state)
 
   auto const& expression_tree_root = expressions.back();
 
-  // Execute benchmark
-  for (auto _ : state) {
-    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
+  state.exec(nvbench::exec_tag::sync, [&](nvbench::launch&) {
+    flush_device_L2_cache(rmm::cuda_stream_view{state.get_cuda_stream().get_stream()});
     cudf::compute_column(table, expression_tree_root);
-  }
+  });
 
   // Use the number of bytes read from global memory
-  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * state.range(0) *
-                          (tree_levels + 1) * sizeof(key_type));
-}
-
-static void CustomRanges(benchmark::internal::Benchmark* b)
-{
-  auto row_counts       = std::vector<cudf::size_type>{100'000, 1'000'000, 10'000'000, 100'000'000};
-  auto operation_counts = std::vector<cudf::size_type>{1, 5, 10};
-  for (auto const& row_count : row_counts) {
-    for (auto const& operation_count : operation_counts) {
-      b->Args({row_count, operation_count});
-    }
-  }
+  state.add_global_memory_reads(static_cast<int64_t>(state.get_summaries().size()) * table_size *
+                                (tree_levels + 1) * sizeof(key_type));
 }
 
 #define AST_TRANSFORM_BENCHMARK_DEFINE(name, key_type, tree_type, reuse_columns, nullable) \
-  BENCHMARK_TEMPLATE_DEFINE_F(AST, name, key_type, tree_type, reuse_columns, nullable)     \
-  (::benchmark::State & st)                                                                \
+  static void name(::nvbench::state& st)                                                   \
   {                                                                                        \
-    BM_ast_transform<key_type, tree_type, reuse_columns, nullable>(st);                    \
+    ::BM_ast_transform<key_type, tree_type, reuse_columns, nullable>(st);                  \
   }                                                                                        \
-  BENCHMARK_REGISTER_F(AST, name)                                                          \
-    ->Apply(CustomRanges)                                                                  \
-    ->Unit(benchmark::kMillisecond)                                                        \
-    ->UseManualTime();
+  NVBENCH_BENCH(name)                                                                      \
+    .set_name(#name)                                                                       \
+    .add_int64_axis("TableSize", {100'000, 1'000'000, 10'000'000, 100'000'000})            \
+    .add_int64_axis("TreeLevels", {1, 5, 10})
```

Comment:
nit: please check if other nvbench benchmarks use CamelCase or snake_case for axis names and align to that. AFAIK, nvbench uses snake_case.

Reply:
Good question -- thanks for asking that. I am not sure off the top of my head.